src/main/java/com/thoughtworks/gauge/datastore/SuiteDataStore.java

```diff
@@ -38,12 +40,20 @@ public static synchronized Object remove(Object key) {
      */
     public static synchronized Object get(Object key) {
         if (key != null) {
-            return map.get().get(key);
+            return getMap().get(key);
         }
         return null;
     }
 
+    private static synchronized ConcurrentHashMap<Object, Object> getMap() {
+        return map.get();
+    }
+
+    public static synchronized Set<Map.Entry<Object, Object>> entrySet() {
```

Comment:
Ah Ok. 

From https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html#entrySet--

```
public Set<Map.Entry<K,V>> entrySet()
Returns a Set view of the mappings contained in this map. 
The set is backed by the map, so changes to the map are reflected in the set, 
and vice-versa. If the map is modified while an iteration over the set is in progress
 (except through the iterator's own remove operation, or through the setValue 
operation on a map entry returned by the iterator) the results of the iteration are 
undefined. The set supports element removal, which removes the corresponding
 mapping from the map, via the Iterator.remove, Set.remove, removeAll, retainAll 
and clear operations. It does not support the add or addAll operations.
```

The new DataStore was introduced to prevent modification of the map.
To get over this it's better to return an immutable Map
https://docs.oracle.com/javase/9/core/creating-immutable-lists-sets-and-maps.htm#JSCOR-GUID-4F3E2B7D-CE90-4862-A78A-414FC08DA6E4

You can call that method `items`

Reply:
I've now updated to return a immutable Map.